### PR TITLE
io.schema: Remove repo URL from revision IDs

### PR DESCRIPTION
**Pull request by spbnick**

### CODE REVIEW

**Comment by spbnick** [comment]
@Inakimalerba, heads up!

**Comment by spbnick** [comment]
@gctucker, do you think you could take a look at this?

File: kcidb/io/schema/v3.py [hunk 1]
```diff
@@ -26,8 +26,7 @@
 
 # A regular expression pattern matching strings containing revision IDs
 REVISION_ID_PATTERN = \
-    f"{GIT_REPOSITORY_URL_PATTERN}" \
-    f"@{GIT_REPOSITORY_COMMIT_HASH_PATTERN}" \
+    f"{GIT_REPOSITORY_COMMIT_HASH_PATTERN}" \
```
Comment (gctucker) [inline kcidb/io/schema/v3.py:29]: It's not new in this PR, but maybe `GIT_COMMIT_HASH_PATTERN` would be better, i.e. without `REPOSITORY` for the same reason as with the `git_commit_hash` field.

Comment (spbnick) [reply kcidb/io/schema/v3.py:29]: Yes, makes sense, I could do that as part of the fix for #85.

**Comment by spbnick** [comment]
> So, what happens if a same git sha1 gets tested in 2 different trees? Do they both have a revision entry in the database with the same id but different repo URLs?

Yes. Similar to how we would have duplicate revision entries if two CI systems discovered the same revision.

**Comment by spbnick** [comment]
@gctucker, do you think this can go in, or would you like me to change something?

**Comment by spbnick** [comment]
> I'm not entirely sure of the consequences of having different database entries with the same ID, but that's not specific to this PR.
Yeah, we already have to deal with that, and it certainly will be "interesting" :grin:.
Thank you, merging!

**Pull request merged**